src/hal/lib/hal_comp.c

```diff
@@ -372,36 +372,39 @@ const char *hal_comp_name(int comp_id)
 
 // use only for owner_ids of pins, params or functs
 // may return NULL if buggy using code
+// This was a bottleneck with rcomps validating pins
+// Move checks so that instances get checked first
+
 hal_comp_t *halpr_find_owning_comp(const int owner_id)
 {
-    hal_comp_t *comp = halpr_find_comp_by_id(owner_id);
-    if (comp != NULL)
-	return comp;  // legacy case: the owner_id refers to a comp
+hal_comp_t *comp;
 
-    // nope, so it better be an instance
+    // more likely to be instcomp, so test that first
     hal_inst_t *inst = halpr_find_inst_by_id(owner_id);
-    if (inst == NULL) {
-	HALERR("BUG: owner_id %d refers neither to a hal_comp_t nor an hal_inst_t",
-	       owner_id);
-	return NULL;
-    }
 
+    if (inst == NULL) { // is it a legacy comp?
+        comp = halpr_find_comp_by_id(owner_id);
+        if (comp != NULL)
```

Comment:
Can we please don't use statements without curly braces. Here are a few good reasons:
http://stackoverflow.com/questions/2125066/is-it-bad-practice-to-use-an-if-statement-without-brackets
MISRA-C also forbids it: https://dev.eclipse.org/sonar/rules/show/squid:S00121?modal=true&layout=false